### ISSUE TITLE
Naming items doesn't work

### ISSUE THREAD

**Issue by ShaneBeee** [SkriptLang/Skript#1505]
I noticed Bensku added 1.13 to the main branch so im going to post my findings.
(Skript version 2.3-alpha1)

It appears that skript isn't parsing aliases as items properly
example:
```
command /test:
	trigger:
		give player 1 of diamond sword named "TEST"
```
or 
```
command /test:
	trigger:
		set {_test} to diamond sword
		set display name of {_test} to "TEST"
		give player 1 of {_test}
```
Both of these just give said player a diamond sword with its default name
But if I set {_test} to a diamond sword, as text parsed as an item:
```
command /test:
	trigger:
		set {_test} to "diamond sword" parsed as item
		set display name of {_test} to "TEST"
		give player 1 of {_test}
```
This will give the player a diamond sword with a name

Same goes for enchantments and lore.

**Comment by iOshawott** [comment]
yea, it's very uncomfortable >.<

**Comment by iOshawott** [comment]
p.s.
unenchanted pickaxe is "diamond pickaxe" and "if player's tool is diamond pickaxe": works
enchanted pickaxe is "DIAMOND_PICKAXE" and "if player's tool is diamond pickaxe": doesn't works
"if player's tool is any pickaxe" also broken - doesn't work anyway

**Comment by bensku** [comment]
The parse issue appears to not be solved, though I'm not sure if it could also be an user error. Regardless, naming items will now work if the syntax is recognized.

**Comment by ShaneBeee** [comment]
Hey, So I see it said it was fixed. But it seems to have gotten worse ;)
If I do this
```
command /test:
	trigger:
		give player 1 of diamond sword named "TEST"
```
It now won't even give me an item.
if I remove the named part, it gives me a diamond sword. But with the name it gives me nothing.

**Comment by Nicofisi** [comment]
@ShaneBeee yeah, @Pikachu920 has told bensku about that an hour ago, thanks for reporting here though 👍

**Comment by ShaneBeee** [comment]
@Nicofisi  You're welcome :)

**Comment by RiverThrough** [comment]
After  2.3-alpha4 you can name items but if you add lore same issue happens

**Comment by GiraffeCubed** [comment]
> 
> 
> After 2.3-alpha4 you can name items but if you add lore same issue happens

Can confirm, I can't spawn items with lore. [Gist](https://gist.github.com/GiraffeCubed/37616618d652e8399643a2bc7eb2a373)

**Comment by Nicofisi** [comment]
> https://gist.github.com/GiraffeCubed/37616618d652e8399643a2bc7eb2a373

This is due to the incompatibility of skQuery with the latest version of Skript. @GiraffeCubed if you use something from skQuery that's not available in other addons please make a feature request about it here in the Skript's issue tracker, or talk about this to some addon dev - skQuery has been abandoned, unsupported and not recommended to use by pretty much everyone for quite a while.

@TheLimeGlass to repeat, if you plan to keep developing your fork of skQuery, please delete the ExprLoredItemStack class altogether

**Comment by Nicofisi** [comment]
@RiverThrough could you exactly please describe what happens to you and with what code

**Comment by RiverThrough** [comment]
> 
> 
> @RiverThrough could you please exactly describe what happens to you and with what code

Oops sorry, I had completely forgotten the "with lore" syntax was part of skquery and not vanilla skript, the issue is not for to report then my apologies

**Comment by Nicofisi** [comment]
Can someone confirm that this works in vanilla Skript with no addons, please <3

**Comment by ShaneBeee** [comment]
> Can someone confirm that this works in vanilla Skript with no addons, please <3

@Nicofisi here is what I tested (no addons, fresh install of skript alpha 4)
```
command /test:
	trigger:
		give player 1 of diamond sword named "&aTEST"
		give player 1 of diamond sword named "&aTest" with lore "&bLore Test"
```
This gives me 1 diamond sword named TEST and also a diamond sword (unnamed) but the lore is correct "Lore Test"
First sword:
![](https://i.imgur.com/9RQUus8.png)
Second Sword:
![](https://i.imgur.com/XY2hOAk.png)

**Comment by ShaneBeee** [comment]
Another test
```
command /test:
	trigger:
		set {_sword} to diamond sword named "&aTest"
		set 1st line of lore of {_sword} to "&bLore Test"
		give player 1 of {_sword}
```
Outcome:
![](https://i.imgur.com/DnDSS4O.png)
